doc/examples/segmentation/plot_boundary_merge.py

```diff
@@ -3,7 +3,15 @@
 Hierarchical Merging of Region Boundary RAGs
 ============================================
 
-TODO: Description
+This examples demonstrates how to perform hierarchical merging on region
```

Comment:
Based on #2078 we must avoid to start the first sentence by "In this example..." because it will be used as a preview. So the first sentence must be a comprehensive summary of the purpose of this example. I think it's good to start now :)